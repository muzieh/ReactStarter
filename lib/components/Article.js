import React from 'react';
import PropTypes from 'prop-types';
import storeProvider from './storeProvider';

const styles = {
	article: {
		paddingBottom: 10,
		borderBottomStyle: 'solid',
		borderBottomColor: '#aaa',
		borderBottomWidth: 1,
		marginBottom: 10,
	},
	title: {
		fontWeight: 'bold',
	},
	date: {
		fontSize: '0.85em',
		color: '#888',
	},
	author: {
		padding: '10px 0',
	},
	body: {
		paddingLeft: 20,
	},
};

const dateDisplay = dateString => new Date(dateString).toDateString();


const Article = props => {
	const { article, author } = props;
	article.data = null;
	return (
		<div style={styles.article}>
			<div style={styles.title}>{article.title}</div>
			<div style={styles.date}>{dateDisplay(article.date)}</div>
			<div style={styles.author}>
				<a href={author.website}>
					{author.firstName}
					{author.lastName}
				</a>
			</div>
			<div style={styles.body}>{article.body}</div>
		</div>
	);
};

/*
const ArticleContainer = (props, { store }) => {
	return <Article {...props} store={store} />;
};

ArticleContainer.contextTypes = {
	store: PropTypes.object,
};
*/

Article.propTypes = {
	article: PropTypes.shape({
		date: PropTypes.string.isRequired,
		title: PropTypes.string.isRequired,
		body: PropTypes.string.isRequired
	}),
	author: PropTypes.object,
	store: PropTypes.object
};

function extraProps(store, oryginalProps) {
	return {
		author: store.lookupAuthor(oryginalProps.article.authorId),
	};
}

export default storeProvider(extraProps)(Article);
