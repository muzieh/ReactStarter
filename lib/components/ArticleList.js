/* eslint arrow-body-style: ["off", "always"] */
import React from 'react';
import Article from './Article';

const ArticleList = props => {
	const { articles, store } = props;
	return (
		Object.values(articles).map(article => {
			return (
				<Article
					key={article.id}
					article={article}
				/>
			);
		})
	);
};


export default ArticleList;
