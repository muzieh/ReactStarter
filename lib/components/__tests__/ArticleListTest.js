import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ArticleList from '../ArticleList';
import Article from '../Article';

configure({ adapter: new Adapter() });

describe('ArticleList', () => {
	const testProps = {
		articles: {
			a: {
				id: 'a',
				date: 'Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)',
				title: 'title1',
				body: 'body 1',
			},
			b: {
				id: 'b',
				date: 'Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)',
				title: 'title2',
				body: 'body 2',
			},
		},
	};

	it('renders correctly', () => {
		const wrapper = shallow(
			<ArticleList
				{...testProps}
			/>
		);

		expect(wrapper.find(Article)).toHaveLength(2);
		expect(shallowToJson(wrapper)).toMatchSnapshot();
		expect(1).toBe(1);
	});
});
