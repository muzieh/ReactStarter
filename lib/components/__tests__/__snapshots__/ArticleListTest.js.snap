// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
Array [
  <Article
    article={
      Object {
        "body": "body 1",
        "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
        "id": "a",
        "title": "title1",
      }
    }
    key="a"
  />,
  <Article
    article={
      Object {
        "body": "body 2",
        "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
        "id": "b",
        "title": "title2",
      }
    }
    key="b"
  />,
]
`;
